### DISCUSSION TITLE
Automatically prompt signature on wallet change

### DISCUSSION THREAD

**Discussion by 0xAskar** [rainbow-me/rainbowkit#941] [Q&A]
Hi, I was wondering if there was a way to enable an automatic signature for the user if they switched wallets inside their wallet provider. This seems to be a common problem for users on sites with required signature verification, i.e. they switch wallets, **Rainbowkit shows new wallets but doesn't require signature**, then they are confused why they aren't able to access the appropriate features with the new wallet.

Any idea of how to incorporate on my own or if Rainbowkit has a method to do so? Was also wondering why Rainbowkit doesn't automatically prompt the user.?

Thanks,
Askar

**Comment by wong2** [comment]
Yes, I'm here to open the same discussion. If the app is using signature based auth like SIWE, then this is definitely a problem, we need some best practices.

**Reply by 0xAskar** [reply]
Would it make sense that when the user disconnects to erase the user's local storage / cookies so that it reprompts them? Not sure why it isn't prompted automatically, I'm assuming there's something I'm probably missing under the hood that makes it difficult to do so?
